### ISSUE TITLE
Remove `generateNonRootItem()` method

### ISSUE THREAD

**Issue by ivan-015** [Parabeac/Parabeac-Core#36]
## Change Proposal
The method `generateNonRootItem()` is duplicate code of the method `_generateScreen()`. Since the recent attribute refactor allows us to easily iterate through every IntermediateNode's attribute, we can use this to remove the `generateNonRootItem()` method.

### Module and Current Solution
This change will likely affect all the Services, as they directly refer to an IntermediateNode's attribute instead of iterating through them. Currently, `generateNonRootItem()`'s purpose is to manually run a node through each service by iterating through its specific attributes..

## Benefits of the Change
This refactor will allow Egg developers to not have to worry about running a node through our services. In addition, it will shorten the code in Parabeac-core.

**Comment by haardikdharma10** [comment]
Hey @ivan-015, I'd like to take up this issue. Can you roughly help me to get started? I see the duplicate methods are in interpret.dart file wherein `generateNonRootItem` takes a DesignNode as a parameter and `_generateScreen` takes a SketchPageItem as a parameter. What essentially do we need to do next?
@siliconivan, can you help me out with this?

**Comment by ivan-015** [comment]
Hi @haardikdharma10 thanks for showing interest in this issue!
Unfortunately there's been a mistake with this issue, I am currently working on a refactor that will change the way attributes of all intermediateNodes will be read. The refactor would allow us to solve this issue. However, we have not yet merged the refactor into dev or stable.

For the time being, I will close this issue. However, once the refactor is merged into dev we will re-open it and I can better explain the details of this issue. Apologies for the misunderstanding and thanks again!
